src/libraries/Common/tests/Tests/System/IO/PathInternal.Tests.cs

```diff
@@ -64,345 +64,5 @@ public void GetCommonPathLength(string first, string second, bool ignoreCase, in
-        };
-
-        [Theory,
-            MemberData(nameof(RemoveRelativeSegmentsDeviceData)),
-            MemberData(nameof(RemoveRelativeSegmentsDeviceRootData))]
-        [PlatformSpecific(TestPlatforms.Windows)]
-        public void RemoveRelativeSegmentsDeviceTest(string path, int skip, string expected)
-        {
-            Assert.Equal(expected, PathInternal.RemoveRelativeSegments(path, skip));
-            StringBuilder sb = new StringBuilder(expected);
-            sb.Replace('.', '?', 0, 4);
-            expected = sb.ToString();
-
-            sb = new StringBuilder(path);
-            sb.Replace('.', '?', 0, 4);
-            path = sb.ToString();
-            Assert.Equal(expected, PathInternal.RemoveRelativeSegments(path, skip));
-        }
-
-        public static TheoryData<string, int, string> RemoveRelativeSegmentUnixData => new TheoryData<string, int, string>
-        {
-            { "/tmp/home", 1, "/tmp/home" },
-            { "/tmp/..", 1, "/" },
-            { "/tmp/home/../././", 1, "/tmp/" },
-            { "/tmp/../../../", 1, "/" },
-            { "/tmp//home", 1, "/tmp/home" },
-            { "/./tmp//home", 1, "/tmp/home" },
-            { "/../tmp/home", 1, "/tmp/home" },
-            { "/../../../tmp/./home", 1, "/tmp/home" },
-            { "//tmp///home", 1, "/tmp/home" },
-            { "/tmp/home/git/./.././git/runtime/../", 1, "/tmp/home/git/" },
-            { "/./tmp/home", 1, "/tmp/home" },
-
-            { "/tmp/home", 4, "/tmp/home" },
-            { "/tmp/..", 4, "/tmp" },
-            { "/tmp/home/../././", 4, "/tmp/" },
-            { "/tmp/../../../", 4, "/tmp/" },
-            { "/tmp//home", 4, "/tmp/home" },
-            { "/./tmp//home", 2, "/./tmp/home" },
-            { "/../tmp/home", 3, "/../tmp/home" },
-            { "/../../../tmp/./home", 4, "/../tmp/home" },
-            { "//tmp///home", 5, "//tmp/home" },
-            { "/tmp/home/git/./.././git/runtime/../", 5, "/tmp/home/git/" },
-            { "/./tmp/home", 3, "/./tmp/home" },
-
-            { "/tmp/../../", 8, "/tmp/../" },
-            { "/tmp/home/../././", 10, "/tmp/home/" },
-            { "/tmp/../../../", 8, "/tmp/../" },
-            { "/tmp//home/.././/", 11, "/tmp//home/" },
-            { "/./tmp//home/git/git", 7, "/./tmp/home/git/git" },
-            { "/../tmp/./home", 8, "/../tmp/home" },
-            { "/../../../tmp/./home", 8, "/../../../tmp/home" },
-            { "//tmp///home/..", 5, "//tmp/" },
-            { "/tmp/home/git/./.././git/runtime/../", 16, "/tmp/home/git/./git/" },
-            { "/./tmp/home/././", 7, "/./tmp/home/" },
-        };
-
-        [Theory,
-            MemberData(nameof(RemoveRelativeSegmentUnixData))]
-        [PlatformSpecific(TestPlatforms.AnyUnix)]
-        public void RemoveRelativeSegmentsUnix(string path, int skip, string expected)
-        {
-            Assert.Equal(expected, PathInternal.RemoveRelativeSegments(path, skip));
-        }
```

Comment:
I'll bring all these back and simply change the called API, so we only have incremental tests.

Reply:
After taking another look, I realized I would not be able to bring them back without considerably changing the test cases, because the original expectation was that the internal method would be told how many characters should be skipped from the beginning of the path.
The new method does the detection of the characters to skip automatically.
I will leave these unit tests removed. I double checked the original results and the new APIs would be able to handle them with the same results.

Reply:
It's also important to mention that the internal method was being used by `GetFullPath`. Now it consumes the new method, and the GetFullPath tests are still passing.